src/BodyContent.js

```diff
@@ -1,24 +1,37 @@
-// BodyContent.js
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import Spinner from './Spinner';
 import Navbar from './Navbar';
 import LeftSidebar from './LeftSidebar';
 import RightSidebar from './RightSidebar';
 import ContentBody from './ContentBody';
-import './BodyContent.css'; // Ensure you have the corresponding CSS file
+import './BodyContent.css';
 
 const BodyContent = () => {
-    return (
-        <div className="body-content">
-            
-            <Navbar />
-            <ContentBody/>
-            <div className="content">
-                <LeftSidebar />
-                <RightSidebar />
-            </div>
-           
-        </div>
-    );
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    // Simulate a data fetch or loading
+    setTimeout(() => {
+      setLoading(false);
+    }, 3000); // Adjust time as needed
+  }, []);
+
+  return (
+    <div className="body-content">
+      {loading ? (
+        <Spinner />
+      ) : (
+        <>
+          <Navbar />
+          <ContentBody />
+          <div className="content">
+            <LeftSidebar />
+            <RightSidebar />
+          </div>
```

Comment:
BodyContent.js is the file name and it has a child called ContentBody?

Please refactor the above code by:

1. Create a wrapper called MainWrapper.js
2. It should contain the loading state
3. it should use Navbar,BodyContnet,Left and Right Sidbars

This way when you read it will be more easy for future you and for others to undertsand what you did here